sw/device/tests/entropy_src_fw_override_test.c

```diff
@@ -0,0 +1,407 @@
+                                               .low_threshold = 0}));
+  }
+
+  // Enable FW override.
+  TRY(dif_entropy_src_fw_override_configure(
+      &entropy_src,
+      (dif_entropy_src_fw_override_config_t){
+          .entropy_insert_enable = true,
+          .buffer_threshold = kEntropySrcFifoThreshold,
+      },
+      kDifToggleEnabled));
+
+  // Enable entropy_src.
+  LOG_INFO("Enabling entropy source");
+  TRY(dif_entropy_src_configure(
+      &entropy_src,
+      (dif_entropy_src_config_t){
+          .fips_enable = true,
+          .route_to_firmware = false,
+          .bypass_conditioner = bypass_conditioner,
+          .single_bit_mode = single_bit_mode,
+          .health_test_threshold_scope = false,
+          .health_test_window_size = kEntropySrcHealthTestWindowSize,
+          .alert_threshold = UINT16_MAX},
+      kDifToggleEnabled));
+
+  // Enable the interrupt handler to provide CSRNG with entropy.
+  irq_external_ctrl(true);
+
+  // Enable CSRNG
+  LOG_INFO("Enabling CSRNG");
+  TRY(dif_csrng_configure(&csrng));
+
+  // Enable EDNs in auto request mode
+  LOG_INFO("Enabling EDNs");
+  TRY(dif_edn_set_auto_mode(&edn0, edn_params0));
+  TRY(dif_edn_set_auto_mode(&edn1, edn_params1));
+
+  LOG_INFO("Entropy complex configured");
+
+  return OK_STATUS();
+}
+
+/**
+ * Configure the entropy source in extract and insert mode and run some entropy
+ * consumers.
+ */
+status_t firmware_override_extract_insert(
+    dif_entropy_src_single_bit_mode_t single_bit_mode,
+    bool bypass_conditioner) {
+  LOG_INFO("==================");
+  LOG_INFO("Configuring single_bit_mode=%u, bypass_conditioner=%d",
+           single_bit_mode, bypass_conditioner);
+
+  entropy_config(single_bit_mode, bypass_conditioner);
+
+  // Generate some random numbers.
+  uint32_t data;
+  for (size_t i = 0; i < 16; i++) {
+    TRY(rv_core_ibex_testutils_get_rnd_data(&rv_core_ibex, kTimeoutUsec,
+                                            &data));
+  }
+
+  // Run an AES encryption and decryption process.
+  dif_aes_transaction_t transaction = {
+      .operation = kDifAesOperationEncrypt,
+      .mode = kDifAesModeEcb,
+      .key_len = kDifAesKey256,
+      .key_provider = kDifAesKeySoftwareProvided,
+      .mask_reseeding = kDifAesReseedPerBlock,
+      .manual_operation = kDifAesManualOperationAuto,
+      .reseed_on_key_change = false,
+      .ctrl_aux_lock = false,
+  };
+
+  TRY(aes_testutils_setup_encryption(transaction, &aes));
+  AES_TESTUTILS_WAIT_FOR_STATUS(&aes, kDifAesStatusOutputValid, true,
+                                kTimeoutUsec);
+  TRY(aes_testutils_decrypt_ciphertext(transaction, &aes));
+
+  // Run KMAC
+  dif_kmac_key_t software_key = {
+      .share0 = {0x43424140, 0x47464544, 0x4B4A4948, 0x4F4E4D4C, 0x53525150,
+                 0x57565554, 0x5B5A5958, 0x5F5E5D5C},
+      .share1 = {0},
+      .length = kDifKmacKeyLen256,
+  };
+  uint32_t output[8];
+
+  TRY(kmac_testutils_config(&kmac, false));
```

Comment:
The default KMAC config will not use any entropy from EDN. This is unfortunate. Is it possible to make the following changes here?
```
.entropy_mode = kDifKmacEntropyModeEdn,
.entropy_fast_process = false,
.entropy_hash_threshold = 1,
.msg_mask = true,
```